drivers/ieee802154/ieee802154_stm32wba.c

```diff
@@ -0,0 +1,1051 @@
+	}
+}
+
+static void stm32wba_receive_failed(stm32wba_802154_rx_error_t error)
+{
+	const struct device *dev = stm32wba_get_device();
+
+	enum ieee802154_rx_fail_reason reason;
+
+	if (error == STM32WBA_802154_RX_ERROR_NO_BUFFERS) {
+		reason = IEEE802154_RX_FAIL_NOT_RECEIVED;
+	} else {
+		reason = IEEE802154_RX_FAIL_OTHER;
+	}
+
+	if (IS_ENABLED(CONFIG_IEEE802154_STM32WBA_LOG_RX_FAILURES)) {
+		LOG_INF("Receive failed, error = %d", error);
+	}
+
+	stm32wba_data.last_frame_ack_fpb = false;
+	stm32wba_data.last_frame_ack_seb = false;
+
+	if (stm32wba_data.event_handler) {
+		stm32wba_data.event_handler(dev, IEEE802154_EVENT_RX_FAILED, (void *)&reason);
+	}
+}
+
+#if defined(CONFIG_NET_L2_CUSTOM_IEEE802154)
+static int stm32wba_configure_extended(enum ieee802154_stm32wba_config_type type,
+				       const struct ieee802154_stm32wba_config *config)
+{
+	switch (type) {
+	case IEEE802154_STM32WBA_CONFIG_CCA_THRESHOLD:
+		LOG_DBG("Setting CCA_THRESHOLD: %d", config->cca_thr);
+		if (st_ral_set_cca_energy_detect_threshold(config->cca_thr)) {
+			return -EIO;
+		}
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_CONTINUOUS_RECEPTION:
+		LOG_DBG("Setting CONTINUOUS_RECEPTION: %u", config->en_cont_rec);
+		st_ral_set_continous_reception(config->en_cont_rec);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_MAX_FRAME_RETRIES:
+		LOG_DBG("Setting MAX_FRAME_RETRIES: %u", config->max_frm_retries);
+		st_ral_set_max_frame_retries(config->max_frm_retries);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_MAX_CSMA_FRAME_RETRIES:
+		LOG_DBG("Setting MAX_CSMA_FRAME_RETRIES: %u", config->max_csma_frm_retries);
+		st_ral_set_max_csma_frame_retries(config->max_csma_frm_retries);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_MIN_CSMA_BE:
+		LOG_DBG("Setting MIN_CSMA_BE: %u", config->min_csma_be);
+		st_ral_set_min_csma_be(config->min_csma_be);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_MAX_CSMA_BE:
+		LOG_DBG("Setting MAX_CSMA_BE: %u", config->max_csma_be);
+		st_ral_set_max_csma_be(config->max_csma_be);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_MAX_CSMA_BACKOFF:
+		LOG_DBG("Setting MAX_CSMA_BACKOFF: %u", config->max_csma_backoff);
+		st_ral_set_max_csma_backoff(config->max_csma_backoff);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_IMPLICIT_BROADCAST:
+		LOG_DBG("Setting IMPLICIT_BROADCAST: %u", config->impl_brdcast);
+		st_ral_set_implicitbroadcast(config->impl_brdcast);
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_ANTENNA_DIV:
+		LOG_DBG("Setting ANTENNA_DIV: %u", config->ant_div);
+		if (st_ral_set_ant_div_enable(config->ant_div)) {
+			return -EIO;
+		}
+		break;
+
+	case IEEE802154_STM32WBA_CONFIG_RADIO_RESET:
+		LOG_DBG("Setting RADIO_RESET");
+		if (st_ral_radio_reset()) {
+			return -EIO;
+		}
+		break;
+
+	default:
+		LOG_ERR("Unsupported configuration type: %d", type);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int stm32wba_attr_get_extended(enum ieee802154_stm32wba_attr attr,
+				      const struct ieee802154_stm32wba_attr_value *value)
```

Comment:
It seems strange a `..._get...()` function with such an output argument declares the argument pointer as pointer to const. I believe it should not and the several casts below should be removed.